freqtrade/rpc/telegram.py

```diff
@@ -262,19 +291,52 @@ def _format_sell_msg(self, msg: Dict[str, Any]) -> str:
                                    ' / {profit_fiat:.3f} {fiat_currency})').format(**msg)
         else:
             msg['profit_extra'] = ''
-
+        msg['currency'] = msg['pair'].split('/')[0]
         message = ("{emoji} *{exchange}:* Selling {pair} (#{trade_id})\n"
-                   "*Profit:* `{profit_ratio:.2%}{profit_extra}`\n"
+                   "*Unrealized Profit:* `{profit_ratio:.2%}{profit_extra}`\n"
                    "*Buy Tag:* `{buy_tag}`\n"
                    "*Sell Reason:* `{sell_reason}`\n"
                    "*Duration:* `{duration} ({duration_min:.1f} min)`\n"
-                   "*Amount:* `{amount:.8f}`\n"
+                   "*Amount:* `{amount:.8f}` {currency}\n"
                    "*Open Rate:* `{open_rate:.8f}`\n"
                    "*Current Rate:* `{current_rate:.8f}`\n"
                    "*Close Rate:* `{limit:.8f}`").format(**msg)
 
         return message
 
+    def _format_sell_fill_msg(self, msg: Dict[str, Any]) -> str:
+        import pprint
+        pprint.pprint(msg)
+        msg['amount'] = round(msg['amount'], 8)
+        msg['profit_percent'] = round(msg['profit_ratio'] * 100, 2)
+        msg['duration'] = msg['close_date'].replace(
+            microsecond=0) - msg['open_date'].replace(microsecond=0)
+        msg['duration_min'] = msg['duration'].total_seconds() / 60
+
+        msg['buy_tag'] = msg['buy_tag'] if "buy_tag" in msg.keys() else None
+        msg['emoji'] = self._get_sell_emoji(msg)
+
+        # Check if all sell properties are available.
+        # This might not be the case if the message origin is triggered by /forcesell
+        if (all(prop in msg for prop in ['gain', 'fiat_currency', 'stake_currency'])
+                and self._rpc._fiat_converter):
+            msg['profit_fiat'] = self._rpc._fiat_converter.convert_amount(
+                msg['profit_amount'], msg['stake_currency'], msg['fiat_currency'])
+            msg['profit_extra'] = (' ({gain}: {profit_amount:.8f} {stake_currency}'
+                                   ' / {profit_fiat:.3f} {fiat_currency})').format(**msg)
+        else:
+            msg['profit_extra'] = ''
+        msg['currency'] = msg['pair'].split('/')[0]
+        message = ("{emoji} *{exchange}:* Sold {pair} (#{trade_id})\n"
+                   "*Profit:* `{profit_ratio:.2%}{profit_extra}`\n"
+                   "*Buy Tag:* `{buy_tag}`\n"
+                   "*Sell Reason:* `{sell_reason}`\n"
+                   "*Duration:* `{duration} ({duration_min:.1f} min)`\n"
+                   "*Amount:* `{amount:.8f}` {currency}\n"
```

Comment:
looking at this, it seems awfully close to the format we use for regular sell messages (the whole method, actually).

Maybe try to combine the two by providing a parameter to `_format_sell_msg` - and only change what's really different (like the header) ... (obviously the same should work for the buy-side as well).
otherwise we'll have these messages "drift apart" over time - as future editors will potentially (for sure?) not remember to change both messages.

Reply:
Yeah they are almost identical, I actually initially started with just modifying the main def's but then broke out into 2 as I thought that is what would be preferred...guess not hah... Will combine back together

Reply:
90% of the content is identical - having 2 separate functions will not make sense, then ;)